tests/mpi/broadcast_03.cc

```diff
@@ -0,0 +1,98 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2021 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+
+
+// Test Utilities::MPI::broadcast() for built-in types.
+
+#include <deal.II/base/mpi.h>
+
+#include <iomanip>
+
+#include "../tests.h"
+
+void
+check(const int int_value)
+{
+  const auto my_proc = Utilities::MPI::this_mpi_process(MPI_COMM_WORLD);
+
+  int buffer;
+  if (my_proc == 0)
+    {
+      buffer = int_value;
+    }
+
+  const auto result = Utilities::MPI::broadcast(MPI_COMM_WORLD, buffer);
+
+  deallog << result << " ";
+  deallog << std::endl;
+}
+
+
+void
+check(const double double_value)
+{
+  const auto my_proc = Utilities::MPI::this_mpi_process(MPI_COMM_WORLD);
+
+  double buffer;
+  if (my_proc == 0)
+    {
+      buffer = double_value;
+    }
+
+  const auto result = Utilities::MPI::broadcast(MPI_COMM_WORLD, buffer);
+
+  deallog << std::setprecision(4) << result << " ";
+  deallog << std::endl;
+}
+
+
+void
+check(const std::complex<double> &complex_value)
+{
+  const auto my_proc = Utilities::MPI::this_mpi_process(MPI_COMM_WORLD);
+
+  std::complex<double> buffer;
+  if (my_proc == 0)
+    {
+      buffer = complex_value;
+    }
+
+  const auto result = Utilities::MPI::broadcast(MPI_COMM_WORLD, buffer);
+
+  deallog << std::setprecision(2) << result << " ";
+  deallog << std::endl;
+}
+
+
+int
+main(int argc, char *argv[])
+{
+  Utilities::MPI::MPI_InitFinalize mpi_initialization(argc, argv, 1);
+  MPILogInitAll                    log;
+
+  const unsigned int root = 0;
+  const bool         on_root =
+    (Utilities::MPI::this_mpi_process(MPI_COMM_WORLD) == root);
+
+  // Test broadcast with int, double, and std::complex data
+  // types. Only provide the right value on the root, and some other
+  // value on non-root processes. We should only see the root value in
+  // the output.
```

Comment:
I don't think this is necessary, because you are not using the value on the other ranks anyways.

Reply:
Not necessary, I just wanted to make sure that the value is overwritten on the other processes. The output file contains output from all processes, so the test does check that the other ranks get the right value.

Reply:
I don't understand. You are already making sure this is the case in the ``if (my_proc == 0)`` statement in the check function. I would replace your code in main with ``check(23);`` etc.